src/value_serializer.rs

```diff
@@ -317,183 +346,224 @@ impl<'a, 's> ValueSerializerHeap<'a, 's> {
+        cast_to_ptr(self.get_cxx_value_serializer()),
         transfer_id,
         array_buffer,
       )
     };
   }
+
+  fn set_treat_array_buffer_views_as_host_objects(&self, mode: bool) {
+    unsafe {
+      v8__ValueSerializer__SetTreatArrayBufferViewsAsHostObjects(
+        cast_to_ptr(self.get_cxx_value_serializer()),
+        mode,
+      )
+    };
+  }
 }
 
 impl ValueSerializerHelper for CxxValueSerializer {
-  fn get_cxx_value_serializer(&mut self) -> &mut CxxValueSerializer {
+  fn get_cxx_value_serializer(&self) -> &CxxValueSerializer {
     self
   }
 }
 
-impl<'a, 's> ValueSerializerHelper for ValueSerializerHeap<'a, 's> {
-  fn get_cxx_value_serializer(&mut self) -> &mut CxxValueSerializer {
-    &mut self.cxx_value_serializer
+impl<'a> ValueSerializerHelper for ValueSerializerHeap<'a> {
+  fn get_cxx_value_serializer(&self) -> &CxxValueSerializer {
+    &self.cxx_value_serializer
   }
 }
 
-impl<'a, 's> ValueSerializerHelper for ValueSerializer<'a, 's> {
-  fn get_cxx_value_serializer(&mut self) -> &mut CxxValueSerializer {
-    &mut self.value_serializer_heap.cxx_value_serializer
+impl<'a> ValueSerializerHelper for ValueSerializer<'a> {
+  fn get_cxx_value_serializer(&self) -> &CxxValueSerializer {
+    &self.value_serializer_heap.cxx_value_serializer
   }
 }
 
-pub struct ValueSerializer<'a, 's> {
-  value_serializer_heap: Pin<Box<ValueSerializerHeap<'a, 's>>>,
+pub struct ValueSerializer<'a> {
+  value_serializer_heap: Pin<Box<ValueSerializerHeap<'a>>>,
+  // ValueSerializerHeap is already !Send and !Sync
+  // but this is just making it explicit
+  _phantom: std::marker::PhantomData<*mut ()>,
+}
+
+impl<'a> crate::cppgc::GarbageCollected for ValueSerializer<'a> {
+  fn trace(&self, visitor: &crate::cppgc::Visitor) {
+    self.value_serializer_heap.trace(visitor);
+  }
 }
 
 /// ValueSerializer is a stack object used as entry-point for an owned and
 /// pinned heap object ValueSerializerHeap.
 /// The 'a lifetime is the lifetime of the ValueSerializerImpl implementation.
 /// The 's lifetime is the lifetime of the HandleScope which is used to retrieve
 /// a Local<'s, Context> for the CallbackScopes
-impl<'a, 's> ValueSerializer<'a, 's> {
+impl<'a> ValueSerializer<'a> {
   pub fn new<D: ValueSerializerImpl + 'a>(
-    scope: &mut HandleScope<'s>,
+    scope: &mut HandleScope,
     value_serializer_impl: Box<D>,
   ) -> Self {
+    let context = scope.get_current_context();
     // create dummy ValueSerializerHeap 'a, and move to heap + pin to address
-    let mut value_serializer_heap = Box::pin(ValueSerializerHeap {
-      value_serializer_impl,
-      cxx_value_serializer: CxxValueSerializer {
-        _cxx_vtable: CxxVTable(std::ptr::null()),
-      },
-      cxx_value_serializer_delegate: CxxValueSerializerDelegate {
-        _cxx_vtable: CxxVTable(std::ptr::null()),
-      },
-      buffer_size: 0,
-      context: scope.get_current_context(),
-    });
+    let value_serializer_heap_ptr =
```

Comment:
Previously this was violating aliasing rules because of the deref through the `Box`. To avoid this, use raw pointers for initialization.